### PR TITLE
KV-quant foundation: benchmark gate + validated modes + mask fix (DAR-313)

### PR DESCRIPTION
**Pull request by Gajesh2007**
Part of epic **DAR-312**. Foundation/tooling only — **no serving-path behavior change**.

## What
- `kv-quant-gate` benchmark CLI + `KVQuant` modules: trustworthy **generation-fidelity** quality gate, **capacity** metric (kv-bytes/token), PPL/logits/output/NIAH suites.
- `kv-attn-selftest`: deterministic numerical proofs (quantized attention ≡ dequantize+SDPA), scorer-mechanism probe.
- First-class validated KV-quant modes (`k8v8:g128`, `k6v6:g64`, …); removed temporary env-var sweep knobs.
- **Mask-fill bug fix** in `quantizedScaledDotProductAttention` (submodule `mlx-swift-lm`): masked positions used `+leastNormalMagnitude` (~0) instead of `-inf`.

## Submodule
Bumps `libs/mlx-swift-lm` → `darkbloom/kvquant` @ `f40442f` (mask fix only). Fork branch pushed to `Layr-Labs/mlx-swift-lm`; upstream-into-main tracked in DAR-321.

## Validation
`swift build` + KVQuant unit tests green; `kv-attn-selftest` all OK.

Tickets: DAR-313, DAR-315.

---
<a href="https://app.blacksmith.sh/Layr-Labs/codesmith/d-inference/pr/377"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1784234647&installation_id=133247490&pr_number=377&repository=Layr-Labs%2Fd-inference&return_to=https%3A%2F%2Fgithub.com%2FLayr-Labs%2Fd-inference%2Fpull%2F377&signature=8c7d8eec34572748d63ec46daa956ba90edc0434dfb27d7e120511858383b1b6"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

### CODE REVIEW

File: provider-swift/Sources/ProviderCore/Benchmark/KVQuant/KVQuantEvaluator.swift [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+                    return "\(issue.kind.rawValue) at \(issue.path.path): \(detail)"
+                }
+                return "\(issue.kind.rawValue) at \(issue.path.path)"
+            }
+            return "local model snapshot is not ready to load: \(issueList.joined(separator: "; "))"
+        case .tooFewTokens(let count):
+            return "need at least two tokens to score sequence, got \(count)"
+        }
+    }
+}
```
Comment (ethenotethan) [inline provider-swift/Sources/ProviderCore/Benchmark/KVQuant/KVQuantEvaluator.swift:378]: 🟡 **[MEDIUM]** 🧩 unused variable assignment in reportMetricValues

💡 _Suggestion:_ remove the unused `_ = modelID` assignment or use the variable if needed

📊 _Score: 3×4 = 12 · Category: dead code_

File: provider-swift/Sources/ProviderCore/Benchmark/KVQuant/KVQuantQualityRunners.swift [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+        }
+
+        let metrics: [String: KVQuantMetricSummary] = [
+            "niah.reference_exact_match_rate": .init(unit: "rate", samples: referenceExact),
+            "niah.exact_match_rate": .init(unit: "rate", samples: candidateExact),
+            "niah.exact_match_agreement_rate": .init(unit: "rate", samples: exactAgree),
+            "niah.reference_contains_rate": .init(unit: "rate", samples: referenceContains),
+            "niah.answer_contains_rate": .init(unit: "rate", samples: candidateContains),
+            "niah.contains_agreement_rate": .init(unit: "rate", samples: containsAgree),
+            "niah.evaluated_cases": .init(unit: "cases", samples: [Double(evaluated)]),
```
Comment (ethenotethan) [inline provider-swift/Sources/ProviderCore/Benchmark/KVQuant/KVQuantQualityRunners.swift:217]: 🟡 **[MEDIUM]** 🧩 unused variable assignment in outputReport

💡 _Suggestion:_ remove the unused `_ = modelID` assignment or use the variable if needed

📊 _Score: 3×4 = 12 · Category: dead code_

File: provider-swift/Sources/ProviderCore/Benchmark/KVQuant/KVQuantThresholdEvaluator.swift [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                            refPeak > 0
+                        {
+                            values["memory.peak_memory_ratio"] = candPeak / refPeak
+                            values["memory.memory_saved_pct"] = (1 - candPeak / refPeak) * 100
+                        }
+                    }
+                }
+            }
+        }
+        return values
```
Comment (ethenotethan) [inline provider-swift/Sources/ProviderCore/Benchmark/KVQuant/KVQuantThresholdEvaluator.swift:119]: 🟡 **[MEDIUM]** 🧩 unused variable assignment in reportMetricValues

💡 _Suggestion:_ remove the unused `_ = modelID` assignment or use the variable if needed

📊 _Score: 3×4 = 12 · Category: dead code_

**Review by ethenotethan — CHANGES_REQUESTED** [review]
## Automated Code Review — Layr-Labs/d-inference#

**Verdict:** REQUEST_CHANGES

### Security — ✅ No issues found

### Performance — ✅ No issues found

### Type_diligence — ✅ No issues found

### Additive_complexity — 3 finding(s) (3 blocking)
- 🟡 **[MEDIUM]** `provider-swift/Sources/ProviderCore/Benchmark/KVQuant/KVQuantEvaluator.swift:378` — unused variable assignment in reportMetricValues
  - _Suggestion:_ remove the unused `_ = modelID` assignment or use the variable if needed
- 🟡 **[MEDIUM]** `provider-swift/Sources/ProviderCore/Benchmark/KVQuant/KVQuantQualityRunners.swift:217` — unused variable assignment in outputReport
  - _Suggestion:_ remove the unused `_ = modelID` assignment or use the variable if needed
- 🟡 **[MEDIUM]** `provider-swift/Sources/ProviderCore/Benchmark/KVQuant/KVQuantThresholdEvaluator.swift:119` — unused variable assignment in reportMetricValues
  - _Suggestion:_ remove the unused `_ = modelID` assignment or use the variable if needed

_3 finding(s) total, 3 blocking. Verdict: REQUEST_CHANGES._

_🤖 Automated review by Centaur · DAR-186_
